src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -351,22 +352,16 @@ private static boolean containsSummaryTag(DetailNode javadoc) {
      * @param javadoc javadoc root node.
      * @return first inline tag node or null if no node is found.
      */
-    private static DetailNode getFirstInlineTag(DetailNode javadoc) {
-        DetailNode node = null;
-        final DetailNode[] children = javadoc.getChildren();
-        for (DetailNode child: children) {
-            // If present as a children of javadoc
-            if (child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
-                node = child;
-            }
-            // If nested inside html tag
-            else if (child.getType() == JavadocTokenTypes.HTML_ELEMENT) {
-                node = getInlineTagNodeWithinHtmlElement(child);
-            }
-
-            if (node != null) {
-                break;
-            }
+    private static Optional<DetailNode> getFirstInlineTag(DetailNode javadoc) {
+        // If present as a children of javadoc
+        Optional<DetailNode> node = Arrays.stream(javadoc.getChildren())
+                .filter(child -> child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG)
+                .findFirst();
+        // If nested inside html tag
+        if (!node.isPresent()) {
```

Comment:
It won't work that way.
For this Javadoc
```
<p>@first<p>
@second
```
The code in master will find `@first`, after your PR this will be `@second`.

Reply:
Done. Please check why the wrecker is failing

Reply:
Wercker is failing on `./.ci/wercker.sh no-error-hibernate-search`:

```bash
[WARNING] Rule 1: org.apache.maven.plugins.enforcer.RequireMavenVersion failed with message:
Detected Maven Version: 3.6.3 is not in the allowed range 3.8.1.

```

Reply:
Is this failure related to my change?

Reply:
No, hibernate just bumped their minimum maven version. We will probably need to open an issue in their Jira.

Reply:
@AkMo3 please rebase.